src/neuroconv/datainterfaces/ecephys/openephys/openephyslegacydatainterface.py

```diff
@@ -0,0 +1,71 @@
+"""Authors: Szonja Weigl, Cody Baker."""
+from typing import Optional
+
+from ..baserecordingextractorinterface import BaseRecordingExtractorInterface
+from ....utils import FolderPathType, get_schema_from_method_signature
+
+
+class OpenEphysLegacyRecordingInterface(BaseRecordingExtractorInterface):
+    """Primary data interface for converting legacy Open Ephys data (.continuous files).
+    Uses :py:class:`~spikeinterface.extractors.OpenEphysLegacyRecordingExtractor`."""
+
+    ExtractorName = "OpenEphysLegacyRecordingExtractor"
+
+    @classmethod
+    def get_source_schema(cls):
+        """Compile input schema for the RecordingExtractor."""
+        source_schema = get_schema_from_method_signature(
+            class_method=cls.__init__,
+            exclude=["stream_name", "block_index", "all_annotations"],
+        )
+        source_schema["properties"]["folder_path"][
+            "description"
+        ] = "Path to directory containing OpenEphys legacy files."
+        return source_schema
+
+    def __init__(
+        self,
+        folder_path: FolderPathType,
+        stream_name: Optional[str] = "Signals CH",
+        block_index: Optional[int] = None,
+        all_annotations: Optional[bool] = True,
+        verbose: bool = True,
+    ):
+        """
+        Initialize reading of OpenEphys legacy recording (.continuous files).
+        See :py:class:`~spikeinterface.extractors.OpenEphysLegacyRecordingExtractor` for options.
+
+        Parameters
+        ----------
+        folder_path : FolderPathType
+            Path to OpenEphys directory.
+        stream_name : str, default: "Signals CH"
+            The name of the recording stream.
+            When the recording stream is not specified the channel stream is chosen if available.
+            When channel stream is not available the name of the stream must be specified.
+        block_index : int, default: None
+        all_annotations : bool, default: True
+        stub_test : bool, default: False
+        verbose : bool, default: True
+        """
+
+        super().__init__(
+            folder_path=folder_path,
+            stream_name=stream_name,
+            block_index=block_index,
+            all_annotations=all_annotations,
+            verbose=verbose,
+        )
+
```

Comment:
@CodyCBakerPhD To harness [get_streams](https://github.com/SpikeInterface/spikeinterface/blob/aa4dfebafcce94b71461fc397a8ed4e69c8ddb53/spikeinterface/extractors/neoextractors/neobaseextractor.py#L139-L147) from SI, I can do that after `super().init` like:

```suggestion
        super().__init__(
            folder_path=folder_path,
            stream_name=stream_name,
            block_index=block_index,
            all_annotations=all_annotations,
            verbose=verbose,
        )
        
        stream_names = self.recording_extractor.get_streams(folder_path)

```
but this way we wouldn't know in advance what streams are in the data, or I am still missing a point here?

Reply:
> but this way we wouldn't know in advance what streams are in the data

Right, so if it's a class method

```python
@classmethod
def get_stream_names(cls, folder_path: FolderPathType) -> List[str]:
    from spikeinterface.extractors import OpenEphysLegacyExtractor

    return OpenEphysLegacyExtractor.get_streams(folder_path=folder_path)
```

Then the user can call it ahead of time to see what's available, then we can also double-check during `__init__`

```python
def __init__(self, ..., stream_name):
    available_strreams = self.get_stream_names(folder_path=folder_path)
    assert stream_name in available_streams, f"The selected stream '{stream_name}' is not in the available streams '{available_strreams}'!"
```

which is usually a more informative error than you might get if you just passed it to SI

Reply:
Ah i see, thanks for the clarification!